sway/desktop/render.c

```diff
@@ -1236,6 +1332,21 @@ static void render_floating_container(struct sway_output *soutput,
 		} else if (state->border == B_PIXEL) {
 			render_top_border(soutput, damage, state, colors, deco_data.alpha, deco_data.corner_radius);
 		}
+
+		fx_renderer_close_stenciling(true);
+
+		// render shadow
+		if (con->shadow_enabled
+			&& config->shadow_blur_sigma > 0
+			&& config->shadow_color[3] > 0.0) {
+			struct wlr_box box = { state->x, state->y, state->width, state->height };
+			scale_box(&box, soutput->wlr_output->scale);
+			render_box_shadow(soutput, damage, &box, config->shadow_color,
+					config->shadow_blur_sigma, deco_data.corner_radius,
+					con->current.border_thickness);
+		}
```

Comment:
We should probably also check if there are client side borders (CSD) just like in `render_view`